app/jobs/send_campaign_job.rb

```diff
@@ -29,15 +29,16 @@ def perform(campaign, partner = nil)
     )
 
     users.each do |user|
-      the_match = Match.create(
-        campaign: campaign,
-        campaign_batch: batch,
-        vaccination_center: campaign.vaccination_center,
-        user: user
-      )
-
-      SendMatchSmsJob.perform_later(the_match)
-      SendMatchEmailJob.perform_later(the_match)
+      REDIS_LOCK.lock!("create_match_for_user_id_#{user.id}", 2000) do
```

Comment:
Why not use "SELECT FOR UPDATE SKIP LOCKED" provide by postgres instead of redlock pattern?